### ISSUE TITLE
Projeto Aula 08.05.24

### ISSUE THREAD

**Issue by JairDantas** [JairDantas/Data_Science_GTI#9]

**Comment by JairDantas** [comment]
# Gerando dados fictícios
for produto in produtos:import pandas as pd

# Carregando o dataset
df = pd.read_csv('/content/Planilha sem título - Página1 (3).csv')

print(df.head())

# Definindo os produtos
produtos = ['tênis', 'bonés', 'camisas', 'meias', 'bermudas', 'capas de celular', 'regatas']

# Definindo os meses
meses = ['Janeiro', 'Fevereiro', 'Março', 'Abril', 'Maio', 'Junho',
         'Julho', 'Agosto', 'Setembro', 'Outubro', 'Novembro', 'Dezembro']

# Criando listas vazias para os dados
dados_produtos = []
dados_meses = []
dados_vendas = []
    for mes in meses:
        dados_produtos.append(produto)
        dados_meses.append(mes)
        dados_vendas.append(random.randint(500, 10000))  # vendas entre 500R$ e 10.000R$

# Criando o DataFrame
df = pd.DataFrame({
    'Produto': dados_produtos,
    'Mês': dados_meses,
    'Vendas': dados_vendas
})

print(df)


# Importando as bibliotecas necessárias
import numpy as np
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

# Gerando dados de exemplo
np.random.seed(0)
X = 2 * np.random.rand(100, 1)  # Variável independente
y = 4 + 3 * X + np.random.randn(100, 1)  # Variável dependente

# Dividindo os dados em conjuntos de treinamento e teste
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

# Criando um modelo de regressão linear
model = LinearRegression()

# Treinando o modelo
model.fit(X_train, y_train)

# Fazendo previsões
y_pred = model.predict(X_test)

# Plotando os dados de teste e a linha de regressão
plt.scatter(X_test, y_test, color='blue')
plt.plot(X_test, y_pred, color='red')
plt.title('Regressão Linear Simples')
plt.xlabel('Variável Independente')
plt.ylabel('Variável Dependente')
plt.show()

import numpy as np
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

# Gerando dados de exemplo
np.random.seed(0)
X = 2 * np.random.rand(100, 1)  # Variável independente
y = 4 + 3 * X + np.random.randn(100, 1)  # Variável dependente

# Cálculos estatísticos básicos
media_X = np.mean(X)
mediana_X = np.median(X)
desvio_padrao_X = np.std(X)

media_y = np.mean(y)
mediana_y = np.median(y)
desvio_padrao_y = np.std(y)

print("Estatísticas da variável independente (X):")
print("Média:", media_X)
print("Mediana:", mediana_X)
print("Desvio Padrão:", desvio_padrao_X)

print("\nEstatísticas da variável dependente (y):")
print("Média:", media_y)
print("Mediana:", mediana_y)
print("Desvio Padrão:", desvio_padrao_y)

# Dividindo os dados em conjuntos de treinamento e teste
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

# Criando um modelo de regressão linear
model = LinearRegression()

# Treinando o modelo
model.fit(X_train, y_train)

# Fazendo previsões
y_pred = model.predict(X_test)

# Plotando os dados de teste e a linha de regressão
plt.scatter(X_test, y_test, color='blue')
plt.plot(X_test, y_pred, color='red')
plt.title('Regressão Linear Simples')
plt.xlabel('Variável Independente')
plt.ylabel('Variável Dependente')
plt.show()

import numpy as np
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

# Gerando dados de exemplo
np.random.seed(0)
X = 2 * np.random.rand(100, 1)  # Variável independente
y = 4 + 3 * X + np.random.randn(100, 1)  # Variável dependente

# Dividindo os dados em conjuntos de treinamento e teste
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

# Criando um modelo de regressão linear
model = LinearRegression()

# Treinando o modelo
model.fit(X_train, y_train)

# Fazendo previsões
y_pred = model.predict(X_test)

# Plotando os dados de teste e a linha de regressão
plt.figure(figsize=(10, 5))

# Gráfico de dispersão com a linha de regressão
plt.subplot(1, 2, 1)
plt.scatter(X_test, y_test, color='blue')
plt.plot(X_test, y_pred, color='red')
plt.title('Regressão Linear Simples')
plt.xlabel('Variável Independente')
plt.ylabel('Variável Dependente')

# Histograma das variáveis dependentes (y)
plt.subplot(1, 2, 2)
plt.hist(y, color='green', bins=20)
plt.title('Histograma das Variáveis Dependentes (y)')
plt.xlabel('Valores de y')
plt.ylabel('Frequência')

plt.tight_layout()
plt.show()

# Gráfico de caixa das variáveis independentes (X)
plt.figure(figsize=(6, 5))
plt.boxplot(X)
plt.title('Gráfico de Caixa da Variável Independente (X)')
plt.ylabel('Valores de X')

plt.show()

**Comment by JairDantas** [comment]
@PROFSANTARELLI

**Comment by JairDantas** [comment]
Jair Silva
Patricia Santos
Dario Lucas
